Pre-commit fixes

waynew opened the pull request:
### What does this PR do?

This is not, strictly speaking, necessary. Removing the language version
allows use of one's default supported python (3.6+).

Also consolidates isort config in pyproject.toml and fixes excludes, because the way pre-commit provides black/isort files  on `pre-commit -a` overrides their default exclude settings

### Previous Behavior

Contributors not using Python3.7 in their dev environment would have to add pyenv or similiar

### New Behavior

Now contributors with Python >=3.6 should be A-OK

### Tests written?

Not relevant

### Commits signed with GPG?

Yes

waynew committed:
Remove black py language version

This is not, strictly speaking, necessary. Removing the language version
allows use of one's default supported python (3.6+).

s0undt3ch reviewed (dismissed):
Should be `python3` at least?

DmitryKuzmenko reviewed (dismissed):
Let's try =)

waynew commented:
@s0undt3ch I think you're right - I should go ahead and just set the [`default_language_version`](https://pre-commit.com/#pre-commit-configyaml---top-level) to python3.

waynew committed:
Set default language version

waynew committed:
Fix pre-commit excludes

When pre-commit provides paths to isort/black, their default excludes
are ignored. So we have to tell pre-commit to *also* exclude black and
isort excludes.

waynew committed:
Also install isort[pyproject] for toml dependency

s0undt3ch commented on pyproject.toml:
```diff
@@ -5,3 +5,12 @@ exclude= '''
 )/
+
+
+[tool.isort]
```
This actually makes isort and black fight over imports.
Let's leave the old `.isort.cfg` as it was?

waynew replied:
I should have added the `ensure_newline_before_comments` flag, but I couldn't get it to work correctly :suspect: 

I'm down with switching back, though when there's a version 5 release of isort that has mo' bettah support, I would prefer heading back to pyproject.toml

Also FWIW when toml hits the version 1 release, Python will be including a toml parser in the stdlib :tada:

s0undt3ch replied:
Cool, we'll switch when it lands :)

waynew replied:
Okay, turns out I figured out how to fix it. My bad :upside_down_face:

waynew committed:
Back to .isort.cfg

At least until version 5 of isort is actually released.

waynew commented:
Doh, don't merge yet!

waynew committed:
Fix toml file

waynew commented:
Okay, *now* it should be good.

s0undt3ch commented:
Run it by tests/conftest.py and see if both agree

s0undt3ch commented:
Or at least the same file in https://github.com/saltstack/salt/pull/56460
